include/lsst/daf/base/DateTime.h

```diff
@@ -57,19 +59,27 @@ namespace base {
 
 class DateTime {
 public:
-    enum Timescale { TAI, UTC, TT };
-    enum DateSystem { JD, MJD, EPOCH };
+    enum DateSystem { JD=0, MJD, EPOCH };
+    enum Timescale { TAI=5, UTC, TT };  // use values that do not overlap DateSystem
+                                        // to avoid confusing one for the other in Python
```

Comment:
@RobertLuptonTheGood yes invalid values for the enums trigger an exception. I'm not sure type safety is possible in the Python layer (though I hope it might be with pybind11). In any case I think you can always get around it by casting ints, and making the values distinct catches mistakes if you do that.